### DISCUSSION TITLE
Rich - how to choose spinner?

### DISCUSSION THREAD

**Discussion by robotnumber3** [Textualize/rich#2989] [Q&A]
Hi, everyone. Can someone please help a beginner?

Rich is excellent and will be very useful. When we create a simple progress bar, such as:

```
from rich.progress import track
from time import sleep

def scrape_data():
    sleep(0.1)

for _ in track(range(100), description='[green]Scraping data'):
    scrape_data()
```
 How can I add the code **spinner='name_of_spinner'** to it?

Thanks for your help.

David

**Comment by Charlotte-br560** [comment]
You can specify the spinner style by adding `spinner='name_of_spinner'` to your `track` function. For example:

```
from rich.progress import track
from time import sleep

def scrape_data():
    sleep(0.1)

for _ in track(range(100), description='[green]Scraping data', spinner='name_of_spinner'):
    scrape_data()
```
Also, if you're into web scraping, you might find [Crawlbase](https://crawlbase.com/) helpful for managing your scraping tasks efficiently.
